src/main/java/com/strandls/esmodule/services/impl/ElasticSearchServiceImpl.java

```diff
@@ -546,6 +549,95 @@ public MapResponse rangeSearch(String index, String type, List<MapRangeQuery> qu
 		return querySearch(index, boolQuery, searchParams, geoAggregationField, geoAggegationPrecision);
 	}
 
+	@Override
+	public Map<String, List<Map<String, Object>>> aggregationByDay(String filter) throws IOException {
+
+		BoolQueryBuilder boolQuery = QueryBuilders.boolQuery();
+		TermQueryBuilder authorFilter = QueryBuilders.termQuery("author_id", filter);
+        boolQuery.filter(authorFilter);
+		AggregationBuilder aggregation = null;
+		aggregation = AggregationBuilders.dateHistogram("agg").field("created_on").dateHistogramInterval(DateHistogramInterval.days(1)).format("yyyy-MM-dd");
+		AggregationResponse aggregationResponse = new AggregationResponse();
+		if (aggregation == null)
+				return null;
+
+		SearchSourceBuilder sourceBuilder = new SearchSourceBuilder();
+		if (boolQuery != null)
+			sourceBuilder.query(boolQuery);
+		sourceBuilder.aggregation(aggregation);
+
+		SearchRequest request = new SearchRequest("extended_observation");
+		request.source(sourceBuilder);
+		SearchResponse response = client.search(request, RequestOptions.DEFAULT);
+		Map<String,List<Map<String, Object>>> groupbyday= new LinkedHashMap <String,List<Map<String, Object>>>();
+		Histogram dateHistogram = response.getAggregations().get("agg");
+
+		for (Histogram.Bucket entry : dateHistogram.getBuckets()) {
+			String year = entry.getKeyAsString().substring(0,4);
+			List<Map<String, Object>> yeardata;
+			if(groupbyday.containsKey(year)) {
+				yeardata = groupbyday.get(year);
+			} else {
+				yeardata = new ArrayList<>();
+			}
+
+			Map<String, Object> data = new HashMap<>();
+        	data.put("date", entry.getKeyAsString());  
+        	data.put("value", entry.getDocCount());
+        	yeardata.add(data);
+			groupbyday.put(year, yeardata);
+		}
+
+		return groupbyday;
+	}
+
+	@Override
+	public Map<String, List<Map<String, Object>>> aggregationByMonth(String user) throws IOException {
```

Comment:
can we get this "extended_observation"  form controller , make it more generic like other endpoints?